### PR TITLE
Nexverse Bid Adapter : initial relese

### PR DESCRIPTION
**Pull request by anand-nexverse**
## Type of change

- [x] Bugfix
- [x] Feature
- [x] New bidder adapter  
- [ ] Updated bidder adapter   
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes

- [ ] Does this change affect user-facing APIs or examples documented on http://prebid.org?
- [ ] Other

### CODE REVIEW

**Comment by anand-nexverse** [comment]
updated function build request

**Comment by bretg** [comment]
docs PR https://github.com/prebid/prebid.github.io/pull/5640

File: integrationExamples/gpt/x-domain/creative.html [hunk 1]
```diff
@@ -2,7 +2,7 @@
...
...n,t)=>{const r=n.createElement("iframe");return Object.entries(Object.assign({},t,e)).forEach((([e,n])=>r.setAttribute(e,n))),r}})();function t(e){return!!e.frames.__pb_locator__}window.pbRender=function(r){let o=r.parent;try{for(;o!==r.top&&!t(o);)o=o.parent;t(o)||(o=r.parent)}catch(e){}return function({adId:t,pubUrl:s,clickUrl:i}){const a=new URL(s,window.location).origin;function c(e,n,r){const s=new MessageChannel;s.port1.onmessage=l(r),o.postMessage(JSON.stringify(Object.assign({message:e,adId:t},n)),a,[s.port2])}function d(n){c(e,{event:"adRenderFailed",info:{reason:n?.reason||"exception",message:n?.message}}),n?.stack&&console.error(n)}function l(e){return function(){try{return e.apply(this,arguments)}catch(e){d(e)}}}c("Prebid Request",{options:{clickUrl:i}},(function(o){let s;try{s=JSON.parse(o.data)}catch(e){return}if("Prebid Response"===s.message&&s.adId===t){const t=n(r.document,{width:0,height:0,style:"display: none",srcdoc:`<script>${s.renderer}<\/script>`});t.onload=l((function(){const o=t.contentWindow;o.Promise.resolve(o.render(s,{sendMessage:c,mkFrame:n},r)).then((()=>c(e,{event:"adRenderSucceeded"})),d)})),r.document.body.appendChild(t)}}))}}(window)})();</script>
```
Comment (patmmccann) [inline integrationExamples/gpt/x-domain/creative.html:5]: Please don't edit this file

File: modules/nexverseBidAdapter.js [hunk 2]
```diff
@@ -0,0 +1,447 @@
...
+      }
+
+      // Return the server request
+      return {
+        method: 'POST',
+        url: endpointUrl,
+        data: JSON.stringify(payload),
+        bidRequest: bid,
+        options: {
+          contentType: 'application/json',
```
Comment (patmmccann) [inline modules/nexverseBidAdapter.js:70]: This slows things down and forces a cors options call

File: modules/nexverseBidAdapter.js [hunk 3]
```diff
@@ -0,0 +1,447 @@
...
+        const bidResponse = {
+          requestId: bid.impid,
+          cpm: bid.price,
+          currency: response.cur || 'USD',
+          width: bid.w || 0,
+          height: bid.h || 0,
+          creativeId: bid.crid || bid.id,
+          ttl: 300,
+          netRevenue: true,
+          meta: {},
```
Comment (patmmccann) [inline modules/nexverseBidAdapter.js:111]: You must be able to set adomain

File: modules/nexverseBidAdapter.js [hunk 4]
```diff
@@ -0,0 +1,447 @@
...
+      osVersion = match[1].replace(/_/g, '.');
+    }
+  }
+  return osVersion;
+}
+
+/**
+ * Determines the device model (if possible).
+ * @returns {string} The device model.
+ */
```
Comment (patmmccann) [inline modules/nexverseBidAdapter.js:417]: Please get this from the request, if the request is unsatisfactory, please modify it in core

File: modules/nexverseBidAdapter.js [hunk 5]
```diff
@@ -0,0 +1,447 @@
...
+
+  return openRtbRequest;
+}
+
+/**
+ * Checks if the request is made over a secure connection (HTTPS).
+ * @returns {boolean} True if the connection is secure (HTTPS), false otherwise.
+ */
+function isSecureRequest() {
+  return location.protocol === 'https:';
```
Comment (patmmccann) [inline modules/nexverseBidAdapter.js:315]: This is inaccurate, secure is always true

File: modules/nexverseBidAdapter.js [hunk 6]
```diff
@@ -0,0 +1,447 @@
...
+    id: bidderRequest.auctionId,
+    imp: imp,
+    site: {
+      page: bidderRequest.refererInfo.page,
+      domain: bidderRequest.refererInfo.domain,
+      ref: bidderRequest.refererInfo.ref || '', // Referrer URL
+    },
+    device: {
+      ua: navigator.userAgent,
+      ip: bidderRequest.ip || '123.123.123.123', // IP address (replace with actual)
```
Comment (patmmccann) [inline modules/nexverseBidAdapter.js:257]: What is this fallback

File: modules/nexverseBidAdapter.js [hunk 7]
```diff
@@ -0,0 +1,447 @@
...
+      os: getDeviceOS(),
+      osv: getDeviceOSVersion(),
+      make: navigator.vendor || '',
+      model: getDeviceModel(),
+      connectiontype: getConnectionType(), // Include connection type
+      geo: {
+        lat: bid.params.geoLat || 0,
+        lon: bid.params.geoLon || 0,
+      },
+      language: navigator.language || 'en',
```
Comment (patmmccann) [inline modules/nexverseBidAdapter.js:268]: Get this from the request

File: modules/nexverseBidAdapter.js [hunk 8]
```diff
@@ -0,0 +1,447 @@
...
+    device: {
+      ua: navigator.userAgent,
+      ip: bidderRequest.ip || '123.123.123.123', // IP address (replace with actual)
+      devicetype: getDeviceType(), // 1 = Mobile/Tablet, 2 = Desktop
+      os: getDeviceOS(),
+      osv: getDeviceOSVersion(),
+      make: navigator.vendor || '',
+      model: getDeviceModel(),
+      connectiontype: getConnectionType(), // Include connection type
+      geo: {
```
Comment (patmmccann) [inline modules/nexverseBidAdapter.js:264]: Get this from the request

File: modules/nexverseBidAdapter.js [hunk 9]
```diff
@@ -0,0 +1,447 @@
...
+  },
+};
+
+/**
+ * Builds the OpenRTB 2.5 request payload.
+ *
+ * @param {Object} bid - The bid request object.
+ * @param {Object} bidderRequest - The bidder request object.
+ * @returns {Object|null} The OpenRTB 2.5 request payload or null if missing mandatory parameters.
+ */
```
Comment (patmmccann) [inline modules/nexverseBidAdapter.js:195]: Why not use the utility function available for this

**Review by patmmccann — CHANGES_REQUESTED** [review]
This bidder duplicates a lot of logic in the request object unnecessarily. Please simplify dramatically.

File: integrationExamples/gpt/x-domain/creative.html (hunk 1 above)
Comment (yogeshverse) [reply integrationExamples/gpt/x-domain/creative.html]: Fixed in new PR https://github.com/prebid/Prebid.js/pull/12472

File: modules/nexverseBidAdapter.js (hunk 2 above)
Comment (yogeshverse) [reply modules/nexverseBidAdapter.js]: Fixed in new PR https://github.com/prebid/Prebid.js/pull/12472

File: modules/nexverseBidAdapter.js (hunk 3 above)
Comment (yogeshverse) [reply modules/nexverseBidAdapter.js:89]: Fixed in new PR https://github.com/prebid/Prebid.js/pull/12472

File: modules/nexverseBidAdapter.js (hunk 4 above)
Comment (yogeshverse) [reply modules/nexverseBidAdapter.js]: Fixed in new PR https://github.com/prebid/Prebid.js/pull/12472

File: modules/nexverseBidAdapter.js (hunk 6 above)
Comment (yogeshverse) [reply modules/nexverseBidAdapter.js]: Fixed in new PR https://github.com/prebid/Prebid.js/pull/12472

File: modules/nexverseBidAdapter.js (hunk 8 above)
Comment (yogeshverse) [reply modules/nexverseBidAdapter.js:204]: Fixed in new PR https://github.com/prebid/Prebid.js/pull/12472

**Comment by yogeshverse** [comment]
> This bidder duplicates a lot of logic in the request object unnecessarily. Please simplify dramatically.

Fixed in new PR https://github.com/prebid/Prebid.js/pull/12472

File: integrationExamples/gpt/x-domain/creative.html (hunk 1 above)
Comment (yogeshverse) [reply integrationExamples/gpt/x-domain/creative.html]: Done.

File: modules/nexverseBidAdapter.js (hunk 2 above)
Comment (yogeshverse) [reply modules/nexverseBidAdapter.js]: Done.

File: modules/nexverseBidAdapter.js (hunk 3 above)
Comment (yogeshverse) [reply modules/nexverseBidAdapter.js:91]: Done.

File: modules/nexverseBidAdapter.js (hunk 4 above)
Comment (yogeshverse) [reply modules/nexverseBidAdapter.js]: Done.

File: modules/nexverseBidAdapter.js (hunk 6 above)
Comment (yogeshverse) [reply modules/nexverseBidAdapter.js]: Done.

File: modules/nexverseBidAdapter.js (hunk 7 above)
Comment (yogeshverse) [reply modules/nexverseBidAdapter.js]: Done.

File: modules/nexverseBidAdapter.js (hunk 8 above)
Comment (yogeshverse) [reply modules/nexverseBidAdapter.js:203]: Done.

File: modules/nexverseBidAdapter.js (hunk 9 above)
Comment (yogeshverse) [reply modules/nexverseBidAdapter.js:140]: Done.

**Review by patmmccann — APPROVED** [review]

**Review by patmmccann — APPROVED** [review]

**Pull request merged**